### ISSUE TITLE
Cannot connect to NameServer

### ISSUE THREAD

**Issue by got3nks** [irmen/Pyro4#141]
```
DEBUG:Pyro4.naming:locating the NS: PYRO:Pyro.NameServer@xxx:9999
DEBUG:Pyro4.core:connecting to PYRO:Pyro.NameServer@xxx:9999
ERROR:Pyro4.core:cannot connect: receiving: not enough data
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/Pyro4/naming.py", line 600, in locateNS
    proxy._pyroBind()
  File "/usr/local/lib/python3.5/site-packages/Pyro4/core.py", line 390, in _pyroBind
    return self.__pyroCreateConnection(True)
  File "/usr/local/lib/python3.5/site-packages/Pyro4/core.py", line 506, in __pyroCreateConnection
    msg = message.Message.recv(conn, [message.MSG_CONNECTOK, message.MSG_CONNECTFAIL], hmac_key=self._pyroHmacKey)
  File "/usr/local/lib/python3.5/site-packages/Pyro4/message.py", line 167, in recv
    msg = cls.from_header(connection.recv(cls.header_size))
  File "/usr/local/lib/python3.5/site-packages/Pyro4/socketutil.py", line 451, in recv
    return receiveData(self.sock, size)
  File "/usr/local/lib/python3.5/site-packages/Pyro4/socketutil.py", line 175, in receiveData
    raise err
Pyro4.errors.ConnectionClosedError: receiving: not enough data
Pyro4.errors.NamingError: Failed to locate the nameserver
```

I am able to connect after server reboot. What could be the issue?

**Comment by got3nks** [comment]
Found the reason in the NameServer log.

[2016-12-02 21:08:38.844,Pyro4.threadpoolserver,WARNING] client connection was denied: no free workers, increase server threadpool size

**Comment by irmen** [comment]
Actually, you made me aware of a regression with this. 

You shouldn't have seen the 'not enough data' error but rather 'Pyro4.errors.CommunicationError: connection rejected, reason: no free workers, increase server threadpool size'

 I've fixed this in 9fd25b36a025d586549849965e3dd9271a83caf7
